cmd/resultscollector/main.go

```diff
@@ -226,105 +233,187 @@ func getConfigMap(owner *unstructured.Unstructured, configMapName, filename stri
+		_, err = clientset.CoreV1().ConfigMaps(scapresultsconf.Namespace).Create(confMap)
+
+		if errors.IsAlreadyExists(err) {
+			return nil
+		}
+		return err
+	}, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), maxRetries))
+}
+
+func uploadErrorConfigMap(errorMsg *resultFileContents, exitcode string,
+	scapresultsconf *scapresultsConfig, clientset *kubernetes.Clientset, dynclient dynamic.Interface) error {
+	return backoff.Retry(func() error {
+		fmt.Println("Trying to upload error ConfigMap")
+		openscapScan, err := getOpenSCAPScanInstance(scapresultsconf.ScanName, scapresultsconf.Namespace, dynclient)
+		if err != nil {
+			return err
+		}
+		confMap := getConfigMap(openscapScan, scapresultsconf.ConfigMapName, "error-msg", errorMsg.contents, errorMsg.compressed, exitcode)
+		_, err = clientset.CoreV1().ConfigMaps(scapresultsconf.Namespace).Create(confMap)
+
+		if errors.IsAlreadyExists(err) {
+			return nil
+		}
+		return err
+	}, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), maxRetries))
+}
+
+func handleCompleteSCAPResults(exitcode string, scapresultsconf *scapresultsConfig,
+	clientset *kubernetes.Clientset, dynclient dynamic.Interface) {
+	arfContents, err := readResultsFile(scapresultsconf.ArfFile, scapresultsconf.Timeout, scapresultsconf.Compress)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	xccdfContents, err := readResultsFile(scapresultsconf.XccdfFile, scapresultsconf.Timeout, scapresultsconf.Compress)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+	go func() {
+		err = uploadToResultServer(arfContents, scapresultsconf)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+		fmt.Println("Uploaded to resultserver")
+		wg.Done()
+	}()
+
+	go func() {
+		err = uploadResultConfigMap(xccdfContents, exitcode, scapresultsconf, clientset, dynclient)
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+		fmt.Println("Uploaded ConfigMap")
+		wg.Done()
+	}()
+	wg.Wait()
+}
+
+func handleErrorInOscapRun(exitcode string, scapresultsconf *scapresultsConfig,
+	clientset *kubernetes.Clientset, dynclient dynamic.Interface) {
+	errorMsg, err := readResultsFile(scapresultsconf.CmdOutputFile, scapresultsconf.Timeout, false)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	err = uploadErrorConfigMap(errorMsg, exitcode, scapresultsconf, clientset, dynclient)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	fmt.Println("Uploaded ConfigMap")
+}
+
+func getOscapExitCode(scapresultsconf *scapresultsConfig) string {
+	exitcodeContent, err := readResultsFile(scapresultsconf.ExitCodeFile, scapresultsconf.Timeout, false)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	if len(exitcodeContent.contents) < 1 {
+		fmt.Println("exitcode file was empty")
+		os.Exit(1)
+	}
+
+	return string(exitcodeContent.contents[0])
```

Comment:
Why are you trimming the results? I would understand trimming whitespace, but we check the results for correctness anyway..

Reply:
I merely am removing the newline. Given that what we care for in the exitcode is just a number (which always fits in a byte, since it's UTF-8), I just get that byte, turn it into a string, and use it.

This code had issues before because the value was, for instance, "1\n" instead of just "1"